stateful-functions-docs/docs/overview/execution_model.rst

```diff
@@ -0,0 +1,57 @@
+.. Copyright 2019 Ververica GmbH.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+.. _execution_model:
+
+###########################
+Distributed Execution Model
+###########################
+
+.. contents:: :local:
+
+The Apache Flink Dataflow
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All ``Functions``, ``Ingresses``, ``Egresses`` and ``Routers`` get assembled into a single Flink application, backed by a single dataflow. This is a different approach from regular Apache Flink applications, where each transformation function is an individual operator in the dataflow. In those applications, each operator has dedicated resources in the TaskManagers (i.e. slots, threads, memory) and can only send events to operators downstream. Assembling all functions into a single, fixed-structure Flink job ensures that these do not require pre-reserved resources and allows **Stateful Functions** to realize arbitrary function-to-function messaging. The fixed-structure graph represents the "state and messaging fabric", rather than capturing the topology with which different function types interact.
+
+.. image:: ../_static/images/flink_dataflow_graph.png
+   :width: 50%
+   :align: center
+
+The Apache Flink dataflow executes the **Stateful Functions** application as follows:
+
+* Ingresses and Routers run in `Flink Source Functions <https://ci.apache.org/projects/flink/flink-docs-release-1.9/api/java/org/apache/flink/streaming/api/functions/source/SourceFunction.html>`_, Egresses in `Flink Sink Functions <https://ci.apache.org/projects/flink/flink-docs-release-1.9/api/java/org/apache/flink/streaming/api/functions/sink/SinkFunction.html>`_.
+
+* All functions are executed by the ``Function Dispatcher Operator``. The logical ID of each function acts as the key by which the operator is parallelized. Functions and their state are hence sharded by their logical ID.
+
+* All messages from Routers to functions and from function to function are sent through ``keyBy()`` streams that route the message according to the key (i.e. the logical ID).
+
+* Arbitrary function-to-function messaging is implemented by a feedback loop:
+
+  * Functions send messages downstream, routed by key to the ``Feedback Operator``.
+  * The Feedback Operator is co-located shard-per-shard with the Function Dispatcher and places the messages into an in-memory feedback channel.
```

Comment:
@igalshilman is the feedback channel in memory or does it spill to disk?

Reply:
@sjwiesmam it spills to disk after a threshold was reached.